docs/data_classes.rst

```diff
@@ -0,0 +1,301 @@
+.. _data_classes:
+
+============
+Data Classes
```

Comment:
I worry a little about the use of "data classes" here given that [`dataclasses`](https://docs.python.org/3/library/dataclasses.html) are now a thing in the standard library.

Should we just call them "classes", "objects" or "types"?

Reply:
I think `Data Objects` is fine.  I wanted to use `Data` to distinguish `NDCube`, `NDCubeSequence` and `NDCollection` from the coordinate objects like `Global Coords`, `ExtraCoords` and other used by `ndcube`.  Another option could be `ND Objects`.  So those three are members of the ND family.

Reply:
I think `Data Objects` sounds good